src/pkg/cli/compose/convert.go

```diff
@@ -13,6 +13,33 @@ import (
 	compose "github.com/compose-spec/compose-go/v2/types"
 )
 
+type serviceNameReplacer struct {
+	client                     client.Client
+	nonReplaceServiceNameRegex *regexp.Regexp
+	serviceName                string
+	serviceNameRegex           *regexp.Regexp
```

Comment:
Could be nice to move the regexp stuff into this struct as well, to encapsulate it.